### PR TITLE
[refactor] Improve config upgrade script and add test

### PR DESCRIPTION
**Pull request by ervteng**
### Proposed change(s)

This PR moves the config upgrade script to in-module, makes it more testable, changes the documentation to this new path, and adds a unit test for the script. 

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [x] Code refactor
- [ ] Breaking change
- [x] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: docs/Training-ML-Agents.md [hunk 1]
```diff
@@ -181,8 +181,8 @@ and explains the possible settings for each.
 
 **NOTE:** The configuration file format has been changed from 0.17.0 and onwards. To convert
 an old set of configuration files (trainer config, curriculum, and sampler files) to the new
-format, a script has been provided. Run `python config/upgrade_config.py -h` in your  console
-to see the script's usage.
+format, a script has been provided. Run `python -m mlagents.trainers.upgrade_config -h` in your
+console to see the script's usage.
```
Comment (vincentpierre) [inline docs/Training-ML-Agents.md:185]: I tried to convert an already converted file using the script and got this error : 
```
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/vincentpierre/Documents/ml-agents/ml-agents/mlagents/trainers/upgrade_config.py", line 119, in <module>
    main()
  File "/Users/vincentpierre/Documents/ml-agents/ml-agents/mlagents/trainers/upgrade_config.py", line 99, in main
    behavior_config_dict = convert_behaviors(old_config)
  File "/Users/vincentpierre/Documents/ml-agents/ml-agents/mlagents/trainers/upgrade_config.py", line 21, in convert_behaviors
    trainer_type = config["trainer"]
KeyError: 'trainer'
```
I think we should try catch some key errors and raise more graceful errors.

File: ml-agents/mlagents/trainers/tests/test_config_conversion.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
+import yaml
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/tests/test_config_conversion.py:1]: I tried converting the trainer_config.yaml I think it works well. Only feedback is that I would like some spaces between the different behaviors and curriculum / sampler.

Comment (ervteng) [reply ml-agents/mlagents/trainers/tests/test_config_conversion.py:1]: > A few minor things, but I approve overall.
> Is there a path to deprecation for this file?
> How many releases is it going to stay?

Thanks for the comments, will address before merging.

My thought is 2 releases before deprecation. That should mean the script exists when we publicize the Python CLI (e.g. blog post), but not the release after. Thoughts?

File: docs/Training-ML-Agents.md (hunk 1 above)
Comment (ervteng) [reply docs/Training-ML-Agents.md:185]: Added exceptions for the two potential KeyErrors (`trainer` and `use_recurrent`).

File: ml-agents/mlagents/trainers/upgrade_config.py [hunk 3]
```diff
@@ -46,17 +46,15 @@ def convert_behaviors(old_trainer_config: Dict[str, Any]) -> Dict[str, Any]:
...
 
 
-def write_to_yaml_file(config: Dict[str, Any], output_config: str):
-    unstructed_config = cattr.unstructure(config)
-    unstructed_config = remove_nones(unstructed_config)
+def write_to_yaml_file(unstructed_config: Dict[str, Any], output_config: str) -> None:
     with open(output_config, "w") as f:
         try:
             yaml.dump(unstructed_config, f, sort_keys=False)
         except TypeError:  # Older versions of pyyaml don't support sort_keys
```
[the remark this answers is not in the document]
Comment (ervteng) [reply ml-agents/mlagents/trainers/upgrade_config.py:67]: I think ArgumentError doesn't exist in Python - this does throw a TypeError if we use an older pyyaml.

File: ml-agents/mlagents/trainers/tests/test_config_conversion.py (hunk 2 above)
Comment (vincentpierre) [reply ml-agents/mlagents/trainers/tests/test_config_conversion.py:1]: > will try to find a workaround.

No need to try too hard. If there is not an easy way to do it, leave it. It does not break anything.

Comment (ervteng) [reply ml-agents/mlagents/trainers/tests/test_config_conversion.py:1]: Added a comment about this at the top

**Pull request merged**